src/Handler/Search.hs

```diff
@@ -21,22 +25,52 @@ import qualified Language.PureScript as P
 
 import qualified XMLArrows
 
+resultsPerPage :: Int
+resultsPerPage = 50
+
+maxPages :: Int
+maxPages = 10
+
 getSearchR :: Handler TypedContent
 getSearchR = do
   mquery <- lookupGetParam "q"
+  page <- maybe 1 (max 1) . ((readMaybe . unpack) =<<) <$> lookupGetParam "page"
+
   case mquery of
     Nothing -> redirect HomeR
     Just query -> do
-      results <- case tryParseType query of
-        Just ty | not (isSimpleType ty) -> searchForType ty
-        _ -> searchForName (toLower query)
+      (results, hasMore) <- case tryParseType query of
+        Just ty | not (isSimpleType ty) -> searchForType (min maxPages page) ty
+        _ -> searchForName (min maxPages page) (toLower query)
       selectRep $ do
-        provideRep (htmlOutput query results)
+        provideRep (htmlOutput query results page hasMore)
         provideRep (jsonOutput results)
   where
-    htmlOutput :: Text -> [SearchResult] -> Handler Html
-    htmlOutput query results = do
+    mkPaginationW :: Int -> Bool -> Handler (Maybe Widget)
+    mkPaginationW page hasMore = runMaybeT $ do
+      cr <- MaybeT getCurrentRoute
+      MaybeT $ do
+        urp <- getUrlRenderParams
+        getParams <- Map.fromList . reqGetParams <$> getRequest
+        let mkPageLink p = urp cr $ Map.toList
```

Comment:
So you mean the pages keep growing as you advance? So on page 4 you would see "1 2 3 [4] next", but on page 3 you would only see "1 2 [3] next"?